app/src/main/java/org/oppia/app/profile/ProfileChooserFragmentPresenter.kt

```diff
@@ -109,23 +111,31 @@ class ProfileChooserFragmentPresenter @Inject constructor(
       if (model.profile.pin.isEmpty()) {
         profileManagementController.loginToProfile(model.profile.id).observe(fragment, Observer {
           if (it.isSuccess()) {
-            fragment.requireActivity().startActivity(Intent(fragment.context, HomeActivity::class.java))
+            activity.startActivity(Intent(fragment.context, HomeActivity::class.java))
           }
         })
       } else {
         val pinPasswordIntent = PinPasswordActivity.createPinPasswordActivityIntent(
-          fragment.requireContext(),
+          activity,
           chooserViewModel.adminPin,
           model.profile.id.internalId
         )
-        fragment.requireActivity().startActivity(pinPasswordIntent)
+        activity.startActivity(pinPasswordIntent)
       }
     }
   }
 
   private fun bindAddView(binding: ProfileChooserAddViewBinding, @Suppress("UNUSED_PARAMETER") model: ProfileChooserUiModel) {
     binding.root.setOnClickListener {
-      fragment.requireActivity().startActivity(AdminAuthActivity.createAdminAuthActivityIntent(fragment.requireContext(), chooserViewModel.adminPin, getColor()))
+      if (chooserViewModel.adminPin.isEmpty()) {
+        activity.startActivity(
+          AdminPinActivity.createAdminPinActivity(activity, chooserViewModel.adminProfileId.internalId, getColor())
```

Comment:
This ``createAdminPinActivity`` method seems inconsistent with the one below. I was expecting ``createAdminPinActivityIntent``.

Reply:
Updated.